// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


enum NetworksecurityAlphaAuthorizationPolicyActionEnum {
  NetworksecurityAlphaAuthorizationPolicyActionEnumNO_VALUE_DO_NOT_USE = 0;
  NetworksecurityAlphaAuthorizationPolicyActionEnumACTION_UNSPECIFIED = 1;
  NetworksecurityAlphaAuthorizationPolicyActionEnumALLOW = 2;
  NetworksecurityAlphaAuthorizationPolicyActionEnumDENY = 3;
}

message NetworksecurityAlphaAuthorizationPolicy {
  string name = 1;
  string description = 2;
  string create_time = 3;
  string update_time = 4;
  map<string, string> labels = 5;
  NetworksecurityAlphaAuthorizationPolicyActionEnum action = 6;
  repeated NetworksecurityAlphaAuthorizationPolicyRules rules = 7;
  string project = 8;
  string location = 9;
}

message NetworksecurityAlphaAuthorizationPolicyRules {
  repeated NetworksecurityAlphaAuthorizationPolicyRulesSources sources = 1;
  repeated NetworksecurityAlphaAuthorizationPolicyRulesDestinations destinations = 2;
}

message NetworksecurityAlphaAuthorizationPolicyRulesSources {
  repeated string principals = 1;
  repeated string ip_blocks = 2;
}

message NetworksecurityAlphaAuthorizationPolicyRulesDestinations {
  repeated string hosts = 1;
  repeated int64 ports = 2;
  repeated string methods = 3;
  NetworksecurityAlphaAuthorizationPolicyRulesDestinationsHttpHeaderMatch http_header_match = 4;
}

message NetworksecurityAlphaAuthorizationPolicyRulesDestinationsHttpHeaderMatch {
  string header_name = 1;
  string regex_match = 2;
}

message ApplyNetworksecurityAlphaAuthorizationPolicyRequest {
  NetworksecurityAlphaAuthorizationPolicy resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteNetworksecurityAlphaAuthorizationPolicyRequest {
  string service_account_file = 1;
  NetworksecurityAlphaAuthorizationPolicy resource = 2;
}

message ListNetworksecurityAlphaAuthorizationPolicyRequest {
  string service_account_file = 1;
  string Project = 2;
  string Location = 3;
}

message ListNetworksecurityAlphaAuthorizationPolicyResponse {
  repeated NetworksecurityAlphaAuthorizationPolicy items = 1;
}

service NetworksecurityAlphaAuthorizationPolicyService {
  rpc ApplyNetworksecurityAlphaAuthorizationPolicy(ApplyNetworksecurityAlphaAuthorizationPolicyRequest) returns (NetworksecurityAlphaAuthorizationPolicy);
  rpc DeleteNetworksecurityAlphaAuthorizationPolicy(DeleteNetworksecurityAlphaAuthorizationPolicyRequest) returns (google.protobuf.Empty);
  rpc ListNetworksecurityAlphaAuthorizationPolicy(ListNetworksecurityAlphaAuthorizationPolicyRequest) returns (ListNetworksecurityAlphaAuthorizationPolicyResponse);
}
